qiita_plugins/biom_type/biom_type_plugin/create.py

```diff
@@ -0,0 +1,110 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from os.path import join, basename
+
+from json import loads
+
+from biom import load_table
+from biom.util import biom_open
+from biom.exception import TableException
+from qiita_client import format_payload
+
+
+def create_artifact(qclient, job_id, parameters, out_dir):
+    """Validate and fix a new BIOM artifact
+
+    Parameters
+    ----------
+    qclient : qiita_client.QiitaClient
+        The Qiita server client
+    job_id : str
+        The job id
+    parameters : dict
+        The parameter values to validate and create the artifact
+    out_dir : str
+        The path to the job's output directory
+
+    Returns
+    -------
+    dict
+        The results of the job
+
+    Raises
+    ------
+    ValueError
+        If there is any error gathering the information from the server
+    """
+    prep_id = parameters['template']
+    files = loads(parameters['files'])
+    a_type = parameters['artifact_type']
+
+    if a_type != "BIOM":
+        return format_payload(
+            False, error_msg="Unknown artifact type %s. Supported types: BIOM"
+                             % a_type)
+
+    qclient.update_job_step(job_id, "Step 1: Collecting prep information")
+    prep_info = qclient.get("/qiita_db/prep_template/%s/data" % prep_id)
+    if not prep_info or not prep_info['success']:
```

Comment:
would it make sense for the error handling to happen within `qclient.get`? I suspect this pattern of checking the result will be replicated a reasonable amount?